rfc/006-genpolicy-platforms.md

```diff
@@ -0,0 +1,111 @@
+When a container is started with the tardev-snapshotter, the agent also ensures that the layers are mounted with `dm-verity` protection.
+
+## State of genpolicy
+
+This is a snapshot of the features in Microsoft's fork that are yet to be implemented in upstream Kata: <https://gist.github.com/burgerdev/9330236c209f49c9d97e01f278b3a915>.
+Setting aside the changes for Azure's CSI drivers, the two main categories are YAML support and hardening.
+Both of these are desirable to backport, the difficulty stems from code layout changes between the repositories.
+
+Contrast uses the Microsoft fork of `genpolicy` and its configuration, with a few patches on top.
+
+The upstream policy has support for `dm-verity` calculation, but the corresponding rule is commented out.
+It also uses an outdated and unmaintained version of `tarindex`.
+Upstream will need to introduce support for other snapshotters (that is, Nydus) eventually.
+
+## Proposal
+
+Contrast shouldn't need to package different versions of the `genpolicy` tool or the CLI.
+Instead, the upstream tool should be able to cover generation for all target platforms.
+Contrast should bundle rules and settings for its target platforms, and invoke the tool accordingly.
+
+How do we get there?
+
+### Focus on upstream
+
+The Kata `genpolicy` is already close to the desired state, and will get even more traction once the CoCo community needs policy enforcement.
+We can accelerate progress by porting necessary changes ourselves.
+Also, contributions from Contrast can ensure that the tool becomes flexible enough to support our use cases.
+
+### Prepare to maintain patches
+
+We would like to support a mix-and-match of snapshotters and target platforms, but the CoCo community consensus is to focus on one.
+Thus, it's likely that Contrast will need patches, for example for an updated `tarindex` or even the entire `dm-verity` calculation.
+
+### Develop our own rules and settings
+
+Due to the nature of the settings file, Contrast can't rely on a single vendor to provide this.
+On the other hand, we don't want to force our users to make modifications for our standard target platforms.
+Thus, we develop settings for our known target platforms, validate them through CI and bundle them with the CLI.
+
+The upstream rules should eventually be compatible with all non-tardev platforms.
+This is a shared interest throughout the wider CoCo community, but we can accelerate it through contributions.
+
+For the tardev snapshotter platforms (that is, AKS for now), we start with the Microsoft rule set.
+Once the necessary patches grow uncomfortable (for example reference value divergence), we port the missing parts to a patch for the upstream rules.
+
+## Alternatives considered
+
+### Bundle both Microsoft's and Kata's tool
+
+While this approach seems easiest on the surface, it's going to explode the size of the CLI binary, which is already significant.
```

Comment:
I see the second point, but I think there could be ways around a large CLI (e.g. by sourcing static builds of these tools at runtime, only for the required platform)

Just a note though, not advocating against your proposal